src/main/java/brooklyn/entity/proxy/aws/ElbControllerImpl.java

```diff
@@ -173,175 +174,179 @@ protected void startLoadBalancer() {
-        Integer healthCheckUnhealthyThreshold = getConfig(HEALTH_CHECK_UNHEALTHY_THRESHOLD);
 
         LOG.debug("Re-initialising existing ELB: " + elbName);
 
-        LoadBalancerServiceContext loadBalancerServiceContext = null;
         try {
             loadBalancerServiceContext = newLoadBalancerServiceContext(loc.getIdentity(), loc.getCredential());
-            ELBApi api = api(loadBalancerServiceContext);
 
-            LoadBalancer loadBalancer = api.getLoadBalancerApi().get(elbName);
+            LoadBalancer loadBalancer = getELBApi().getLoadBalancerApi().get(elbName);
             Scheme oldScheme = loadBalancer.getScheme().orNull();
             if (oldScheme != null && !oldScheme.value().equals(elbScheme)) {
                 LOG.warn("Existing ELB {} scheme ({}) is different from configuration ({}); cannot modify; continuing",
                         new Object[] {elbName, oldScheme, elbScheme});
             }
 
-            /*
+            configureSecurityGroups(elbName);
+
+            configureAvailabilityZones(elbName, loadBalancer);
+
+            configureSubnets(elbName, loadBalancer);
+
+            configureListeners(elbName, loadBalancer);
+            configureHealthCheck();
+
+            sensors().set(Attributes.HOSTNAME, loadBalancer.getDnsName());
+            }
+        finally {
+            Closeables2.closeQuietly(loadBalancerServiceContext.unwrap());
+        }
+
+    }
+    private void configureSecurityGroups(String elbName){
+        final Collection<String> securityGroups = getConfig(LOAD_BALANCER_SECURITY_GROUPS);
+
+        if (securityGroups != null && securityGroups.size() > 0) {
+            getELBApi().getLoadBalancerApi().applySecurityGroupsToLoadBalancer(elbName, securityGroups != null ? securityGroups : ImmutableList.<String>of());
+        }
+
+    }
+    private void configureAvailabilityZones(String elbName, LoadBalancer loadBalancer) {
+        Set<String> availabilityZoneNames = (getConfig(AVAILABILITY_ZONES) != null) ? ImmutableSet.copyOf(getConfig(AVAILABILITY_ZONES)) : ImmutableSet.<String>of();
+
+        if (availabilityZoneNames != null && availabilityZoneNames.size() > 0) {
+
             // Set the availability zones
             Set<String> oldAvailabilityZoneNames = loadBalancer.getAvailabilityZones();
             Set<String> oldAvailabilityZoneNamesSet = ImmutableSet.copyOf(oldAvailabilityZoneNames);
-            Set<String> removedAvailabilityZoneNames = Sets.difference(oldAvailabilityZoneNamesSet, availabilityZoneNames);
-            Set<String> addedAvailabilityZoneNames = Sets.difference(availabilityZoneNames, oldAvailabilityZoneNamesSet);
+            Set<String> removedAvailabilityZoneNames = Sets.difference(oldAvailabilityZoneNamesSet, ImmutableSet.copyOf(availabilityZoneNames));
+            Set<String> addedAvailabilityZoneNames = Sets.difference(ImmutableSet.copyOf(availabilityZoneNames), oldAvailabilityZoneNamesSet);
```

Comment:
No point in copying `availabilityZoneNames` again - it's already copied and immutable (from above).